backend/business-partner-agent/src/main/java/org/hyperledger/bpa/security/oauth2/endpoint/endsession/request/KeycloakEndSessionFilter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.security.oauth2.endpoint.endsession.request;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.FilterChain;
+import io.micronaut.http.filter.HttpFilter;
+import io.micronaut.http.netty.NettyMutableHttpResponse;
+import io.micronaut.security.filters.SecurityFilter;
+import io.micronaut.security.oauth2.configuration.OauthConfigurationProperties.OpenIdConfigurationProperties;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+
+@Requires(property = OpenIdConfigurationProperties.PREFIX + ".end-session.keycloak.enabled",
+        notEquals = StringUtils.FALSE)
+@Filter("/**/logout")
+public class KeycloakEndSessionFilter implements HttpFilter {
```

Comment:
Still not sure is this is really needed, see comment below. But if so can you add a comment describing why this class was added?

Reply:
Removing this, the by having the browser call our logout url, the redirects are ok.